pkg/alerting/drivers/config/conversion.go

```diff
@@ -0,0 +1,68 @@
+package config
+
+import (
+	"net/url"
+
+	amCfg "github.com/prometheus/alertmanager/config"
+	commoncfg "github.com/prometheus/common/config"
+	alertingv1 "github.com/rancher/opni/pkg/apis/alerting/v1"
+)
+
+func ToWebhook(endp *alertingv1.WebhookEndpoint) *WebhookConfig {
+	httpConfig := &commoncfg.HTTPClientConfig{}
+	if hc := endp.GetHttpConfig(); hc != nil {
+		if hc.GetAuthorization() != nil {
+			httpConfig.Authorization = &commoncfg.Authorization{
+				Type:        hc.GetAuthorization().GetType(),
+				Credentials: commoncfg.Secret(hc.GetAuthorization().GetCredentials()),
+			}
+		}
+		if hc.GetBasicAuth() != nil {
+			httpConfig.BasicAuth = &commoncfg.BasicAuth{
+				Username: hc.GetBasicAuth().GetUsername(),
+				Password: commoncfg.Secret(hc.GetBasicAuth().GetPassword()),
+			}
+		}
+		if hc.GetOauth2() != nil {
+			httpConfig.OAuth2 = &commoncfg.OAuth2{
+				ClientID:     hc.GetOauth2().GetClientId(),
+				ClientSecret: commoncfg.Secret(hc.GetOauth2().GetClientSecret()),
+				TokenURL:     hc.GetOauth2().GetTokenUrl(),
+			}
+		}
+		if hc.GetProxyUrl() != "" {
+			proxyUrl, err := url.Parse(hc.GetProxyUrl())
+			if err != nil {
+				panic(err)
+				// return validation.Errorf("proxy url must be a valid url : %s", err)
```

Comment:
intentional?

Reply:
yep, I'll remove the comment here, that check is done in `pkg/apis/alerting/v1/validate.go`, so we *should* never get here